ginipaybusiness/src/main/java/net/gini/pay/ginipaybusiness/review/ReviewFragment.kt

```diff
@@ -270,52 +275,59 @@ class ReviewFragment(
             }
         }
     }
-}
 
-private fun GpbFragmentReviewBinding.setKeyboardAnimation(resources: Resources) {
-    if (Build.VERSION.SDK_INT < 30) return
-    ViewCompat.setWindowInsetsAnimationCallback(paymentDetails, object : WindowInsetsAnimationCompat.Callback(DISPATCH_MODE_STOP) {
-        var startBottom = 0
-        var endBottom = 0
-        var startHeight = 0
-        var endHeight = 0
-        override fun onPrepare(animation: WindowInsetsAnimationCompat) {
-            startBottom = paymentDetails.paddingBottom
-            startHeight = pager.height
+    private fun GpbFragmentReviewBinding.removePagerConstraint() {
+        root.post {
+            ConstraintSet().apply {
+                clone(constraintRoot)
+                constrainHeight(R.id.pager, pager.height)
+                clear(R.id.pager, ConstraintSet.BOTTOM)
+                applyTo(constraintRoot)
+            }
         }
+    }
+
+    private fun GpbFragmentReviewBinding.setKeyboardAnimation() {
+        ViewCompat.setWindowInsetsAnimationCallback(paymentDetails, object : WindowInsetsAnimationCompat.Callback(DISPATCH_MODE_STOP) {
+            var startBottom = 0
+            var endBottom = 0
 
-        override fun onStart(
-            animation: WindowInsetsAnimationCompat,
-            bounds: WindowInsetsAnimationCompat.BoundsCompat
-        ): WindowInsetsAnimationCompat.BoundsCompat {
-            endBottom = paymentDetails.paddingBottom
-            endHeight = pager.height
-            paymentDetails.translationY = (endBottom - startBottom).toFloat()
-            pager.pivotY = (resources.getDimension(R.dimen.gpb_page_padding_top) + (ViewCompat.getRootWindowInsets(root)
-                ?.getInsets(windowInsetTypesOf(statusBars = true))?.top?.toFloat() ?: 0f))
-            (startHeight.toFloat() / endHeight).let {
-                pager.scaleX = it
-                pager.scaleY = it
+            override fun onPrepare(animation: WindowInsetsAnimationCompat) {
+                startBottom = paymentDetails.paddingBottom
             }
-            return bounds
-        }
 
-        override fun onProgress(insets: WindowInsetsCompat, runningAnimations: MutableList<WindowInsetsAnimationCompat>): WindowInsetsCompat {
-            runningAnimations.find { it.typeMask == windowInsetTypesOf(ime = true) }?.let { animation ->
-                paymentDetails.translationY = lerp((endBottom - startBottom).toFloat(), 0f, animation.interpolatedFraction)
-                (lerp((startHeight.toFloat() / endHeight), 1f, animation.interpolatedFraction)).let {
-                    pager.scaleX = it
-                    pager.scaleY = it
+            override fun onStart(
+                animation: WindowInsetsAnimationCompat,
+                bounds: WindowInsetsAnimationCompat.BoundsCompat
+            ): WindowInsetsAnimationCompat.BoundsCompat {
+                if (Build.VERSION.SDK_INT >= 30) {
```

Comment:
Could you tell me why only on 30 and above? And what happens on 29 and lower? Thanks!

Reply:
Thank you!